src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -1788,6 +1794,14 @@ static int bitCount(byte a) {
         return Integer.bitCount((int)a & 0xFF);
     }
 
+    static int numberOfTrailingZeros(byte a) {
+        return a != 0 ? Integer.numberOfTrailingZeros(a) : 8;
```

Comment:
Integer.numberOfTrailingZeros((int)a & 0xFF) ?

Reply:
Would it not return incorrect value for a=0.

Reply:
This could be `Integer.numberOfTrailingZeros(a | 0x100);`

Reply:
And I think what he meant was the calculation of the `numberOfLeadingZeros` below

Reply:
> numberOfTrailingZeros

Having a special case of zero will be better from performance perspective.
We may incur one additional comparison on frequent path but given that C2 will intrinsify at a VectorSupport.lanewise level it will not  hurt compiled code performance but will definitely improve slow path code performance.